qiskit_experiments/framework/base_experiment.py

```diff
@@ -223,11 +230,14 @@ def run(
                          ExperimentData container.
         """
 
-        if backend is not None or analysis != "default" or run_options:
+        if backend_run is not None:
+            self._backend_run = backend_run
+
+        if isinstance(run_obj, Backend) or analysis != "default" or run_options:
             # Make a copy to update analysis or backend if one is provided at runtime
             experiment = self.copy()
```

Comment:
With my last suggestion I meant something like this:

```suggestion
        backend_run_changed = backend_run is not None and backend_run != self._backend_run

        if isinstance(run_obj, Backend) or analysis != "default" or run_options or backend_run_changed:
            # Make a copy to update analysis or backend if one is provided at runtime
            experiment = self.copy()
            if backend_run_changed:
                experiment._backend_run = backend_run
```

Reply:
ah got it